libs/designer-ui/src/lib/editor/base/utils/parsesegments.ts

```diff
@@ -298,6 +299,51 @@ export const decodeStringSegmentTokensInLexicalContext = (value: string, nodeMap
 export const encodeStringSegmentTokensInLexicalContext = (value: string, nodeMap: Map<string, ValueSegment>): string =>
   encodeOrDecodeStringSegmentTokens(value, nodeMap, decodeSegmentValueInLexicalContext, encodeSegmentValueInLexicalContext, 'encode');
 
+export const decodeStringSegmentTokensInDomAndLexicalContext = (value: string, nodeMap: Map<string, ValueSegment>): string =>
+  encodeOrDecodeStringSegmentTokensMultipleFunctions(
```

Comment:
I'm wondering if splitting into encodeOrDecodeStringSegmentTokensMultipleFunctions is necessary, I'm not sure if this will get used other than within this context, and instead of using multiple decoders and encoders we could encode/decode logic inside the loop like:
```
const processStringSegmentTokensInDomAndLexicalContext= (
  value: string,
  nodeMap: Map<string, ValueSegment>,
  encode: boolean
): string => {
  return [...nodeMap.keys()].reduce((newValue, key) => {
    const encodedValue = encodeSegmentValueInLexicalContext(encodeSegmentValueInDomContext(key));
    const decodedValue = decodeSegmentValueInLexicalContext(decodeSegmentValueInDomContext(key));
    return newValue.replaceAll(encode ? decodedValue : encodedValue, encode ? encodedValue : decodedValue);
  }, value);
};
```

Reply:
Fair point. Using the decode/encode functions together in a loop probably makes more sense. Making this change now.